GetStream/stream-chat-android
# [3426] Provide fake images for snapshot tests

### 🎯 Goal

We need to get rid of async image loading for snapshot testing.

### 🛠 Implementation details

Providing images for resources via fake `ImageLoader`. 

Unfortunately, due to our complex Coil configuration, was not able to provide proper behavior for UI Components. We should consider using [MockWebServer](https://github.com/coil-kt/coil/blob/main/coil-compose-base/src/androidTest/java/coil/compose/utils/ImageMockWebServer.kt) for that.

### 🎨 UI Changes

| Compose | UI Components |
| --- | --- |
| ![io getstream chat android uitests snapshot compose UserAvatarTest_testChannelAvatar](https://user-images.githubusercontent.com/9600921/165974462-8a7ccd19-d309-449a-95e8-7797710b19c5.png) | ![io getstream chat android uitests snapshot uicomponents UserAvatarTest_testChannelAvatar](https://user-images.githubusercontent.com/9600921/165974479-47187fb8-58f1-4012-8b69-222fa94efec2.png) |
| ![io getstream chat android uitests snapshot compose UserAvatarTest_testUserAvatar](https://user-images.githubusercontent.com/9600921/165974475-c4f368fa-42f4-4ec8-876a-00fac524961f.png) | ![io getstream chat android uitests snapshot uicomponents UserAvatarTest_testUserAvatar](https://user-images.githubusercontent.com/9600921/165974480-89445ad3-a6a0-4f87-9260-8894a9b65bc5.png) |

### 🧪 Testing

1. Execute the command below
```
./gradlew stream-chat-android-ui-uitests:executeScreenshotTests -Precord -Pandroid.testInstrumentationRunnerArguments.filter=io.getstream.chat.android.uitests.util.SnapshotTestFilter
```

2. Check generated screenshots 

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
